### PR TITLE
libvncserver: make systemd support optional

### PR DESCRIPTION
**Pull request by hmenke**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by hmenke** [comment]
Result of `nixpkgs-review pr 230053` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>aqemu</li>
  </ul>
</details>
<details>
  <summary>17 packages built:</summary>
  <ul>
    <li>arcanPackages.all-wrapped</li>
    <li>arcanPackages.arcan</li>
    <li>arcanPackages.arcan-wrapped</li>
    <li>arcanPackages.cat9-wrapped</li>
    <li>arcanPackages.durden-wrapped</li>
    <li>arcanPackages.pipeworld-wrapped</li>
    <li>arcanPackages.prio-wrapped</li>
    <li>arcanPackages.xarcan</li>
    <li>libsForQt5.krdc</li>
    <li>libsForQt5.krfb</li>
    <li>libvncserver</li>
    <li>libvncserver.dev</li>
    <li>remmina</li>
    <li>virt-manager-qt</li>
    <li>virtscreen</li>
    <li>virtscreen.dist</li>
    <li>x11vnc</li>
  </ul>
</details>

File: pkgs/development/libraries/libvncserver/default.nix [hunk 1]
```diff
@@ -8,6 +8,7 @@
 , zlib
 , libgcrypt
 , libpng
+, withSystemd ? stdenv.isLinux
```
Comment (wegank) [inline pkgs/development/libraries/libvncserver/default.nix:11]:
```suggestion
, withSystemd ? lib.meta.availableOn stdenv.hostPlatform systemd
```

File: pkgs/development/libraries/libvncserver/default.nix [hunk 2]
```diff
@@ -29,12 +30,16 @@ stdenv.mkDerivation rec {
     cmake
   ];
 
+  cmakeFlags = [
+    "-DWITH_SYSTEMD=${if withSystemd then "ON" else "OFF"}"
+  ];
+
```
Comment (wegank) [inline pkgs/development/libraries/libvncserver/default.nix:36]: It should probably be moved under `propagatedBuildInputs`. Otherwise, LGTM.

Comment (hmenke) [reply pkgs/development/libraries/libvncserver/default.nix:36]: I've grepped through nixpkgs before writing this and found that most put this either directly before or after `nativeBuildInputs`, but I can of course move it around. It's just not clear to me what is the convention.

Comment (wegank) [reply pkgs/development/libraries/libvncserver/default.nix:36]: I think it's common practice to put `flags` after `buildInput` blocks, and committers do check that, albeit undocumented. Other orders may be less obvious, regarding `preConfigure` and `flags`, for example.

Comment (7c6f434c) [reply pkgs/development/libraries/libvncserver/default.nix:36]: I would not be confident that there is a clear order, but I think there is a preference that «similar» declarations stay together.

**Review by wegank — APPROVED** [review]

**Review by risicle — APPROVED** [review]
`nixpkgs-review` happy, macos 10.15 & nixos x86_64

**Pull request merged**